[CI] RuntimeFieldsClientYamlTestSuiteIT test {yaml=runtime_fields/10_keyword/docvalue_fields} failing

**Build Scans:**
- [elasticsearch-periodic #4026 / openjdk23_checkrestcompat_java-matrix](https://gradle-enterprise.elastic.co/s/dvcj6oxbjvndw)
- [elasticsearch-periodic #4022 / openjdk23_checkrestcompat_java-matrix](https://gradle-enterprise.elastic.co/s/afqmvtyecbvss)

**Reproduction Line:**
```
./gradlew ":modules:runtime-fields-common:yamlRestCompatTest" --tests "org.elasticsearch.painless.RuntimeFieldsClientYamlTestSuiteIT.test {yaml=runtime_fields/10_keyword/fetch fields}" -Dtests.seed=1A5337974592447A -Dtests.locale=ps -Dtests.timezone=America/Argentina/Tucuman -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('test%20%7Byaml%3Druntime_fields%2F10_keyword%2Fdocvalue_fields%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.painless.RuntimeFieldsClientYamlTestSuiteIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [runtime_fields/10_keyword:99]: Expected a list containing
0: expected "Thursday" but was "Thu"
```

**Issue Reasons:**
- [main] 2 consecutive failures in step openjdk23_checkrestcompat_java-matrix
- [main] 2 failures in test test {yaml=runtime_fields/10_keyword/docvalue_fields} (0.6% fail rate in 348 executions)
- [main] 2 failures in step openjdk23_checkrestcompat_java-matrix (66.7% fail rate in 3 executions)
- [main] 2 failures in pipeline elasticsearch-periodic (50.0% fail rate in 4 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/modules/runtime-fields-common/src/yamlRestTest/resources/rest-api-spec/test/runtime_fields/10_keyword.yml
+++ b/modules/runtime-fields-common/src/yamlRestTest/resources/rest-api-spec/test/runtime_fields/10_keyword.yml
@@ -19,7 +19,7 @@ setup:
                 script: |
                   Instant instant = Instant.ofEpochMilli(params._source.timestamp);
                   ZonedDateTime dt = ZonedDateTime.ofInstant(instant, ZoneId.of("UTC"));
-                  emit(dt.dayOfWeek.getDisplayName(TextStyle.FULL, Locale.ROOT));
+                  emit(dt.dayOfWeek.getDisplayName(TextStyle.FULL, Locale.ENGLISH));
               # Test fetching many values
               day_of_week_letters:
                 type: keyword
@@ -218,7 +218,7 @@ setup:
             day_of_week:
               type: keyword
               script: |
-                emit(doc['timestamp'].value.dayOfWeekEnum.getDisplayName(TextStyle.SHORT, Locale.ROOT));
+                emit(doc['timestamp'].value.dayOfWeekEnum.getDisplayName(TextStyle.SHORT, Locale.ENGLISH));
   - do:
       search:
         index: sensor

--- a/modules/runtime-fields-common/src/yamlRestTest/resources/rest-api-spec/test/runtime_fields/13_keyword_calculated_at_index.yml
+++ b/modules/runtime-fields-common/src/yamlRestTest/resources/rest-api-spec/test/runtime_fields/13_keyword_calculated_at_index.yml
@@ -28,7 +28,7 @@ setup:
                 script: |
                   Instant instant = Instant.ofEpochMilli(params._source.timestamp);
                   ZonedDateTime dt = ZonedDateTime.ofInstant(instant, ZoneId.of("UTC"));
-                  emit(dt.dayOfWeek.getDisplayName(TextStyle.FULL, Locale.ROOT));
+                  emit(dt.dayOfWeek.getDisplayName(TextStyle.FULL, Locale.ENGLISH));
               # Test fetching many values
               day_of_week_letters:
                 type: keyword

